### PR TITLE
Feature azureblob discovery

### PR DESCRIPTION
**Pull request by AnasBensalah**

### CODE REVIEW

File: lib/issues/azure_blob_exposed_files.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
+
+module Intrigue
+module Issue
+class AzureBlobExposedFiles < BaseIssue
+
+  def self.generate(instance_details={})
+
+    to_return = {
+      added: "2020-01-01",
```
Comment (shpendk) [inline lib/issues/azure_blob_exposed_files.rb:9]: Date is wrong and should be corrected

File: lib/issues/azure_blob_exposed_files.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+      name: "azure_blob_exposed_files",
+      pretty_name: "Exposed files within an Azure Blob",
+      severity: 1,
+      status: "confirmed",
```
Comment (shpendk) [inline lib/issues/azure_blob_exposed_files.rb:13]: Suggest to change it to potential. Files may be public by design.

File: lib/issues/azure_blob_open_publicly.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
+
+module Intrigue
+module Issue
+class AzureBLOBOpenPublicly < BaseIssue
+
+  def self.generate(instance_details={})
+
+    to_return = {
+      added: "2020-01-01",
```
Comment (shpendk) [inline lib/issues/azure_blob_open_publicly.rb:9]: Date is wrong and should be corrected

File: lib/issues/azure_blob_open_publicly.rb [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+      name: "azure_blob_open_publicly",
+      pretty_name: "Open Azure Blob",
+      severity: 2,
+      status: "confirmed",
```
Comment (shpendk) [inline lib/issues/azure_blob_open_publicly.rb:13]: Suggest to change it to potential. Blob may be open to public by design

File: lib/tasks/search_azure_blob.rb [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+    blob_url = 'blob.core.windows.net'
+
+    # Lists
+    account_brute_list = []
+    valid_account_list = []
+    container_list = []
+
+    # get entity details
+    if entity_type == "Domain"
+      account_name = account_name.split('.', 0)
```
Comment (shpendk) [inline lib/tasks/search_azure_blob.rb:45]: The split actually returns an array. I tested it with a domain and it didn't seem to work, so this looks like a bug.

**Review by shpendk — CHANGES_REQUESTED** [review]
Some changes needed before we can merge this one. Also, there is a file named `search_azure_blob_test.rb`. Please remove it since it seems to be a test file.

File: lib/tasks/search_azure_blob.rb (hunk 5 above)
Comment (AnasBensalah) [reply lib/tasks/search_azure_blob.rb:45]: fixed

File: lib/issues/azure_blob_open_publicly.rb (hunk 3 above)
Comment (AnasBensalah) [reply lib/issues/azure_blob_open_publicly.rb:9]: Date corrected

File: lib/issues/azure_blob_open_publicly.rb (hunk 4 above)
Comment (AnasBensalah) [reply lib/issues/azure_blob_open_publicly.rb:13]: change made

File: lib/issues/azure_blob_exposed_files.rb (hunk 2 above)
Comment (AnasBensalah) [reply lib/issues/azure_blob_exposed_files.rb:13]: change made

**Pull request closed without merging**